### PR TITLE
Swift 3.0 beta 6 (acegreen's branch, but with fixes for OS X)

### PR DESCRIPTION
**Pull request by pixelspark**
As for some reason I can't seem to submit a pull request to @acegreen's branch, I'll submit it here: this is @acegreen's branch for Swift 3 (XCode beta 6), but now with fixes that let it build under OS X. 

Further discussion is here: https://github.com/danielgindi/Charts/pull/1342

### CODE REVIEW

**Comment by liuxuan30** [comment]
@pixelspark hey thanks. Just one question, do you want me merge #1342 first and then this one?

**Comment by liuxuan30** [comment]
I just merged #1342, could you pull and perhaps double check everything good so I can merge this PR?

**Comment by pixelspark** [comment]
@liuxuan30 thanks, I have just rebased my changes atop of your Swift-3.0 branch (a3c03c2). It builds with these changes on OSX & looks fine to me!

File: Charts/Classes/Charts/ChartViewBase.swift [hunk 1]
```diff
@@ -357,7 +357,7 @@ public class ChartViewBase: NSUIView, ChartDataProvider, ChartAnimatorDelegate
                 // 23 is the smallest recommended font size on the TV
                 font = NSUIFont.systemFont(ofSize: 23, weight: UIFontWeightMedium)
             #else
-                font = NSUIFont.systemFont(ofSize: NSUIFont.systemFontSize)
+                font = NSUIFont.systemFont(ofSize: NSUIFont.nsuiSystemFontSize)
```
Comment (liuxuan30) [inline Charts/Classes/Charts/ChartViewBase.swift]: guess it is supposed to start with `NSUI` ?

File: Charts/Classes/Data/Implementations/Standard/ChartDataEntry.swift [hunk 2]
```diff
@@ -51,14 +51,14 @@ public class ChartDataEntry: NSObject
...
     {
-        let object = object as? AnyObject
-        
-        if (object === nil)
-        {
-            return false
-        }
-        
-        if (!object!.isKind(of: type(of: self)))
+		if (object == nil)
```
Comment (liuxuan30) [inline Charts/Classes/Data/Implementations/Standard/ChartDataEntry.swift]: seems weird here

File: Charts/Classes/Highlight/BarChartHighlighter.swift [hunk 3]
```diff
@@ -198,7 +198,7 @@ public class BarChartHighlighter: ChartHighlighter
         self.chart?.getTransformer(ChartYAxis.AxisDependency.left).pixelToValue(&pt)
         let xVal = Double(pt.x)
         
-        let setCount = barData.dataSetCount ?? 0
```
Comment (liuxuan30) [inline Charts/Classes/Highlight/BarChartHighlighter.swift]: seems we need ??

File: Charts/Classes/Utils/ChartUtils.swift [hunk 4]
```diff
@@ -394,7 +394,7 @@ public class ChartUtils
             }
             else
             {
-                newArray.append(val as! NSObject)
+				newArray.append(NSString(string: val!))
```
Comment (liuxuan30) [inline Charts/Classes/Utils/ChartUtils.swift]: indent issue?

File: Charts/Classes/Utils/ChartSelectionDetail.swift [hunk 5]
```diff
@@ -77,30 +77,29 @@ public class ChartSelectionDetail: NSObject
     
     public override func isEqual(_ object: Any?) -> Bool
     {
-
```
Comment (liuxuan30) [inline Charts/Classes/Utils/ChartSelectionDetail.swift]: same as above

File: Charts/Classes/Highlight/ChartHighlight.swift [hunk 6]
```diff
@@ -122,40 +122,39 @@ public class ChartHighlight: NSObject
     
     public override func isEqual(_ object: Any?) -> Bool
     {
-        
-        let object = object as? AnyObject
-        
-        if (object === nil)
+        if (object == nil)
```
Comment (liuxuan30) [inline Charts/Classes/Highlight/ChartHighlight.swift]: not sure which implementation is better

**Comment by liuxuan30** [comment]
@pixelspark I have added some line notes. Please review. BTW, some of the changes has weird indents, maybe you want to double check

File: Charts/Classes/Utils/ChartPlatform.swift [hunk 7]
```diff
@@ -200,6 +200,12 @@ types are aliased to either their UI* implementation (on iOS) or their NS* imple
 		UIGraphicsBeginImageContextWithOptions(size, opaque, scale)
 	}
 
+	extension NSUIFont {
```
Comment (liuxuan30) [inline Charts/Classes/Utils/ChartPlatform.swift]: not sure why, but the indent seems not 4 spaces..

**Comment by pixelspark** [comment]
@liuxuan30 Thanks. Will check indenting, probably an issue with my XCode settings between projects. Re the other remarks: will have a look into those later today or tomorrow.

File: Charts/Classes/Highlight/BarChartHighlighter.swift (hunk 3 above)
Comment (liuxuan30) [inline Charts/Classes/Highlight/BarChartHighlighter.swift]: sorry I missed the warning. no longer need ?? 0 here

**Comment by liuxuan30** [comment]
@pixelspark One more thing, as SE-1107 introduces open, I shall move all public to open, so if possible, could you hold on and migrate it for macOS after I move iOS portion to open? Or I could do it later.

**Comment by pixelspark** [comment]
@liuxuan30 I don't think that I need to make any classes 'open' in addition to what your commit will fix. The OSX specific parts are mainly internal to the Charts framework (i.e. none of the NSUI classes should be exposed to users of the framework, let alone be declared open...)

**Comment by liuxuan30** [comment]
OK, not fully looked into macOS yet, just being careful if it will override access control. It's not only classes but also class members, I have converted all available ones.
Whenever you are ready :)

**Comment by pixelspark** [comment]
@liuxuan30 fixed open/public for OSX (this PR is now based on your latest Swift-3.0 commit, e.g. ce4e180). Feel free to fix any whitespace or code styling issues yourself :)

**Comment by liuxuan30** [comment]
Thanks.

**Pull request merged**